### PR TITLE
IndexedDB: Add tests for event properties

### PR DESCRIPTION
**Pull request by inexorabletash**
Add tests that verify the bubbles and cancelable properties of events fired during open and transaction abort.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5108/)

**Comment by inexorabletash** [comment]
@pwnall - can you review these?

https://github.com/w3c/IndexedDB/issues/86 for context

Note that the test and spec disagree about the cancelable attribute for the event fired from `open()`, but the test passes in Blink and Gecko. Spec says "false", tests/implementations say "true" - I'll fix the spec once this is ready.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/209495231)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 92d411bd5f88a78aa250ad4765804cf542c1c405
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5108/IndexedDB/idbfactory-open-error-properties.html">/IndexedDB/idbfactory-open-error-properties.html</a></summary>


|                    Subtest                     | Results | Messages |
|------------------------------------------------|---------|----------|
|                                                | OK      |          |
| `Properties of error event from failed open()` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5108/IndexedDB/transaction-abort-request-error.html">/IndexedDB/transaction-abort-request-error.html</a></summary>


|                                  Subtest                                   | Results | Messages |
|----------------------------------------------------------------------------|---------|----------|
|                                                                            | OK      |          |
| `Properties of error events fired at requests when aborting a transaction` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/209495232)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 92d411bd5f88a78aa250ad4765804cf542c1c405
Using browser at version 58.0.3029.6 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5108/IndexedDB/idbfactory-open-error-properties.html">/IndexedDB/idbfactory-open-error-properties.html</a></summary>


|                    Subtest                     | Results | Messages |
|------------------------------------------------|---------|----------|
|                                                | OK      |          |
| `Properties of error event from failed open()` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5108/IndexedDB/transaction-abort-request-error.html">/IndexedDB/transaction-abort-request-error.html</a></summary>


|                                  Subtest                                   | Results | Messages |
|----------------------------------------------------------------------------|---------|----------|
|                                                                            | OK      |          |
| `Properties of error events fired at requests when aborting a transaction` | PASS    |          |

</details>
</details>

File: IndexedDB/transaction-abort-request-error.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    request.onerror = t.step_func(e => {
+      assert_equals(request.readyState, 'done',
+                    'Request\'s done flag should be set');
+      assert_equals(request.result, undefined,
+                    'Request\'s result should be undefined');
+      assert_equals(request.error.name, 'AbortError',
+                    'Request\'s error should be AbortError');
+
+      assert_equals(e.target, request, 'event target was request');
```
Comment (pwnall) [inline IndexedDB/transaction-abort-request-error.html]: Why is this "was" rather than the usual "should be"?

**Review by pwnall — APPROVED** [review]
LGTM with one nit.

File: IndexedDB/transaction-abort-request-error.html (hunk 1 above)
Comment (inexorabletash) [inline IndexedDB/transaction-abort-request-error.html]: No reason. Probably just sleepy - I'll fix.

**Pull request merged**